workos/resources/list.py

```diff
@@ -79,10 +81,18 @@ class BaseWorkOsListResource(
     data: Sequence[ListableResource]
     list_metadata: ListMetadataType
 
-    list_method: Callable = Field(exclude=True)
+    list_method: Callable[
+        ...,
+        Union[
+            "WorkOsListResource[ListableResource, ListAndFilterParams, ListMetadataType]",
+            "Awaitable[AsyncWorkOsListResource[ListableResource, ListAndFilterParams, ListMetadataType]]",
+        ],
+    ] = Field(exclude=True)
     list_args: ListAndFilterParams = Field(exclude=True)
 
-    def _parse_params(self):
+    def _parse_params(
+        self,
+    ) -> Tuple[Dict[str, Union[int, str, None]], Dict[str, Any]]:
```

Comment:
Mapping? (unless we need to mutate the result, in which case I think it needs to be a `Dict`)
```suggestion
    ) -> Tuple[Mapping[str, Union[int, str, None]], Mapping[str, Any]]:
```

Reply:
The first item us mutated, but the second isn't. Will change.